weave/pantry.py

```diff
@@ -245,35 +259,30 @@ def does_file_exist(self, file_path, **kwargs):
         ----------
         file_path: str
             Local path to file being checked
-        **load_mongo: bool (default=false)
-            If true, the mongo supplement will be reloaded to get current
-            pantry contents.
+        **FORCE_LOAD_SUPPLEMENT: bool (default=false)
+            If true, the mongo supplement will be reloaded with the ENTIRETY of
+            the current pantry contents.
 
         Returns
         ----------
         list of basket uuids of where the file exists if it does. If file does
         not exist in the pantry, an empty list is returned.
         """
+        if not _HAS_PYMONGO:
+            raise ImportError("Missing Dependency. The package 'pymongo' "
+                              "is required to use this function.")
+
         integrity_data = derive_integrity_data(file_path)['hash']
+        mongo_loader = MongoLoader(pantry=self)
 
-        load_mongo = kwargs.get('load_mongo', False)
+        if kwargs.get("FORCE_LOAD_SUPPLEMENT", False):
+            mongo_loader.load_mongo_supplement(
+                self.index.to_pandas_df(max_rows=None).uuids
+            )
```

Comment:
I think if we are forcing the mongo supplement to be loaded in, shouldn't we generate the index before?

Reply:
I'm gonna say the user should have an up to date index before calling this method. Calling generate index should not be common, ideally they'd have a up to date index already. Same thing applies when we go to create a basket, we assume the index/pantry is already updated.